### ISSUE TITLE
FP2: SIM unlocking screen only displayed for ~1s on startup, SIM cannot be unlocked thereafter

### ISSUE THREAD

**Issue by casimir7** [ubports/ubuntu-touch#1271]
FP2, UT16.04 (OTA10)

### Steps to reproduce
Restart (or start) the device with only one SIM card placed in slot 2 (slot 1 empty)

### Expected behavior
*) After startup I expect the SIM unlocking screen either to stay until I enter the SIM code or cancel the process.
*) SIM unlocking should always be possible when a sim is present, independent of the history

### Actual behavior
About 2 out of 3 times at startup the SIM unlocking screen does not unfold completely or is shown only for 1 second before the ubuntu startup splash screen is displayed again until the locked desktop appears. After that, it is not possible to unlock the SIM in the settings window, as the button remains grayed and inert.

### Logfiles and additional information
I don't know in what log file the problem could appear (please suggest).
The only ERROR entries I found after startup were in 
unity8.log (which have probably occured only after the restart problem and are hence not relevant):
8<------------------------------
[2019-10-26:10:48:35.393] qtmir.applications: Application["com.ubuntu.ubports_ubports"]::Application()
[2019-10-26:10:48:35.394] unity.topsurfacelist: addApplication(com.ubuntu.ubports_ubports)
[2019-10-26:10:48:35.395] unity.topsurfacelist: appendPlaceholder(com.ubuntu.ubports_ubports)
[2019-10-26:10:48:35.473] file:///usr/share/unity8//Stage/Stage.qml:1631:17: QML WindowInfoItem: Binding loop detected for property "maxWidth"
[2019-10-26:10:48:35.485] unity.topsurfacelist: raiseId(id=2) - Model busy (modelState=1). Try again in the next event loop.
[2019-10-26:10:48:35.491] unity.topsurfacelist: appendSurfaceHelper after (index=0,appId=unity8-dash,surface=0x1edef90,id=1),(index=1,appId=com.ubuntu.ubports_ubports,surface=0x0,id=2)
[2019-10-26:10:48:35.494] qtmir.applications: Application["com.ubuntu.ubports_ubports"]::requestFocus() - emitting focusRequested()
[2019-10-26:10:48:35.524] qtmir.sessions: Wakelock acquired "3"
[2019-10-26:10:48:35.530] unity.topsurfacelist: raiseId(id=2) - do it now.
[2019-10-26:10:48:35.530] unity.topsurfacelist: move from=1 to=0
[2019-10-26:10:48:35.534] qtmir.applications: Application["unity8-dash"]::setRequestedState(requestedState=suspended)
[2019-10-26:10:48:35.535] unity.topsurfacelist: move after (index=0,appId=com.ubuntu.ubports_ubports,surface=0x0,id=2),(index=1,appId=unity8-dash,surface=0x1edef90,id=1)
[2019-10-26:10:48:35.706] qtmir.surfaces: MirSurface[0x1edef90,"unity8-dash"]::updateVisibility(false)
[2019-10-26:10:48:35.791] qtmir.mir: SessionAuthorizer::connection_is_allowed - this= SessionAuthorizer(0xb00a6ae4) pid= 5582
[2019-10-26:10:48:35.816] qtmir.mir: SessionAuthorizer::configure_display_is_allowed - this= SessionAuthorizer(0xb00a6ae4) pid= 5582
[2019-10-26:10:48:35.818] qtmir.mir: SessionAuthorizer::set_base_display_configuration_is_allowed - this= SessionAuthorizer(0xb00a6ae4) pid= 5582
[2019-10-26:10:48:35.819] qtmir.mir: SessionAuthorizer::screencast_is_allowed - this= SessionAuthorizer(0xb00a6ae4) pid= 5582
[2019-10-26:10:48:35.820] qtmir.mir: SessionAuthorizer::prompt_session_is_allowed - this= SessionAuthorizer(0xb00a6ae4) pid= 5582
[2019-10-26:10:48:35.821] qtmir.mir: SessionListener::starting - this= SessionListener(0xad4d1f5c) session= 0xa8b110bc
[2019-10-26:10:48:35.834] qtmir.sessions: SessionManager::onSessionStarting - sessionName= Main.qml
[2019-10-26:10:48:35.835] qtmir.applications: Application["com.ubuntu.ubports_ubports"]::setSession(session=qtmir::Session(0x1737c00))
[2019-10-26:10:48:36.460] qtmir.mir: MirWindowManagerImpl::add_surface(): size requested ( 3450 , 2300 ) and placed ( 1080 , 1851 )
[2019-10-26:10:48:36.461] qtmir.mir: SessionListener::surface_created - this= SessionListener(0xad4d1f5c) session= 0xa8b110bc surface= 0xa8b3d138
[2019-10-26:10:48:36.463] qtmir.surfaces: MirSurface[0x2131ff0,"com.ubuntu.ubports_ubports"]::MirSurface()
[2019-10-26:10:48:36.804] unity.topsurfacelist: appendSurface appId=com.ubuntu.ubports_ubports surface=qtmir::MirSurface(0x2131ff0), filling out placeholder. after: (index=0,appId=com.ubuntu.ubports_ubports,surface=0x2131ff0,id=2),(index=1,appId=unity8-dash,surface=0x1edef90,id=1)
[2019-10-26:10:48:36.809] qtmir.surfaces: MirSurface[0x2131ff0,"com.ubuntu.ubports_ubports"]::registerView(35290280) after=1
[2019-10-26:10:48:36.812] qtmir.surfaces: MirSurface[0x2131ff0,"com.ubuntu.ubports_ubports"]::updateVisibility(true)
[2019-10-26:10:48:36.816] qtmir.surfaces: MirSurface[0x1edef90,"unity8-dash"]::setFocused(false)
[2019-10-26:10:48:36.817] qtmir.surfaces: MirSurface[0x2131ff0,"com.ubuntu.ubports_ubports"]::setFocused(true)
[2019-10-26:10:48:36.817] qtmir.surfaces: MirSurface[0x2131ff0,"com.ubuntu.ubports_ubports"]::setKeymap("de+neo")
xkbcommon: ERROR: Key "<MDSW>" added to modifier map for multiple modifiers; Using Mod3, ignoring Mod5
[2019-10-26:10:48:36.879] qtmir.applications: Application["com.ubuntu.ubports_ubports"]::setInternalState(state=Running)
[2019-10-26:10:48:53.644] qtmir.applications: Application["morph-browser"]::Application()
[2019-10-26:10:48:53.644] unity.topsurfacelist: addApplication(morph-browser)
[2019-10-26:10:48:53.644] unity.topsurfacelist: appendPlaceholder(morph-browser)
[2019-10-26:10:48:53.772] file:///usr/share/unity8//Stage/Stage.qml:1631:17: QML WindowInfoItem: Binding loop detected for property "maxWidth"
[2019-10-26:10:48:53.792] unity.topsurfacelist: raiseId(id=3) - Model busy (modelState=1). Try again in the next event loop.
[2019-10-26:10:48:53.799] unity.topsurfacelist: appendSurfaceHelper after (index=0,appId=com.ubuntu.ubports_ubports,surface=0x2131ff0,id=2),(index=1,appId=unity8-dash,surface=0x1edef90,id=1),(index=2,appId=morph-browser,surface=0x0,id=3)
[2019-10-26:10:48:53.800] qtmir.applications: Application["morph-browser"]::setExemptFromLifecycle(true)
[2019-10-26:10:48:53.802] qtmir.applications: Application["morph-browser"]::requestFocus() - emitting focusRequested()
[2019-10-26:10:48:53.806] unity.topsurfacelist: raiseId(id=3) - do it now.
[2019-10-26:10:48:53.807] unity.topsurfacelist: move from=2 to=0
[2019-10-26:10:48:53.812] qtmir.applications: Application["com.ubuntu.ubports_ubports"]::setRequestedState(requestedState=suspended)
[2019-10-26:10:48:53.814] qtmir.applications: Application["com.ubuntu.ubports_ubports"]::suspend()
[2019-10-26:10:48:53.814] qtmir.applications: Application["com.ubuntu.ubports_ubports"]::setInternalState(state=SuspendingWaitSession)
[2019-10-26:10:48:53.817] unity.topsurfacelist: move after (index=0,appId=morph-browser,surface=0x0,id=3),(index=1,appId=com.ubuntu.ubports_ubports,surface=0x2131ff0,id=2),(index=2,appId=unity8-dash,surface=0x1edef90,id=1)
[2019-10-26:10:48:54.001] qtmir.surfaces: MirSurface[0x2131ff0,"com.ubuntu.ubports_ubports"]::updateVisibility(false)
[2019-10-26:10:48:54.159] qtmir.mir: SessionAuthorizer::connection_is_allowed - this= SessionAuthorizer(0xb00a6ae4) pid= 5657
[2019-10-26:10:48:54.174] qtmir.mir: SessionAuthorizer::configure_display_is_allowed - this= SessionAuthorizer(0xb00a6ae4) pid= 5657
[2019-10-26:10:48:54.175] qtmir.mir: SessionAuthorizer::set_base_display_configuration_is_allowed - this= SessionAuthorizer(0xb00a6ae4) pid= 5657
[2019-10-26:10:48:54.175] qtmir.mir: SessionAuthorizer::screencast_is_allowed - this= SessionAuthorizer(0xb00a6ae4) pid= 5657
[2019-10-26:10:48:54.175] qtmir.mir: SessionAuthorizer::prompt_session_is_allowed - this= SessionAuthorizer(0xb00a6ae4) pid= 5657
[2019-10-26:10:48:54.176] qtmir.mir: SessionListener::starting - this= SessionListener(0xad4d1f5c) session= 0xa8b236e4
[2019-10-26:10:48:54.176] qtmir.sessions: SessionManager::onSessionStarting - sessionName= morph-browser
[2019-10-26:10:48:54.176] qtmir.applications: Application["morph-browser"]::setSession(session=qtmir::Session(0x234f9c0))
[2019-10-26:10:48:55.285] qtmir.applications: Application["com.ubuntu.ubports_ubports"]::setInternalState(state=SuspendingWaitProcess)
[2019-10-26:10:48:55.299] qtmir.applications: Application["com.ubuntu.ubports_ubports"]::setInternalState(state=Suspended)
[2019-10-26:10:48:56.158] qtmir.mir: MirWindowManagerImpl::add_surface(): size requested ( 800 , 600 ) and placed ( 1080 , 1851 )
[2019-10-26:10:48:56.159] qtmir.mir: SessionListener::surface_created - this= SessionListener(0xad4d1f5c) session= 0xa8b236e4 surface= 0xa8b41748
[2019-10-26:10:48:56.161] qtmir.surfaces: MirSurface[0x22f0c98,"morph-browser"]::MirSurface()
[2019-10-26:10:48:56.955] unity.topsurfacelist: appendSurface appId=morph-browser surface=qtmir::MirSurface(0x22f0c98), filling out placeholder. after: (index=0,appId=morph-browser,surface=0x22f0c98,id=3),(index=1,appId=com.ubuntu.ubports_ubports,surface=0x2131ff0,id=2),(index=2,appId=unity8-dash,surface=0x1edef90,id=1)
[2019-10-26:10:48:56.956] qtmir.surfaces: MirSurface[0x22f0c98,"morph-browser"]::registerView(29242568) after=1
[2019-10-26:10:48:56.956] qtmir.surfaces: MirSurface[0x22f0c98,"morph-browser"]::updateVisibility(true)
[2019-10-26:10:48:56.956] qtmir.surfaces: MirSurface[0x2131ff0,"com.ubuntu.ubports_ubports"]::setFocused(false)
[2019-10-26:10:48:56.957] qtmir.surfaces: MirSurface[0x22f0c98,"morph-browser"]::setFocused(true)
[2019-10-26:10:48:56.957] qtmir.surfaces: MirSurface[0x22f0c98,"morph-browser"]::setKeymap("de+neo")
xkbcommon: ERROR: Key "<MDSW>" added to modifier map for multiple modifiers; Using Mod3, ignoring Mod5
[2019-10-26:10:48:57.037] qtmir.applications: Application["morph-browser"]::setInternalState(state=Running)
[2019-10-26:10:53:36.368] qtmir.sensor: Screen::toggleSensors - enable= false
[2019-10-26:10:53:36.516] This backend doesn't support multiple users
[2019-10-26:10:53:36.575] qtmir.applications: Application["morph-browser"]::setRequestedState(requestedState=suspended)
[2019-10-26:10:53:36.576] qtmir.applications: Application["morph-browser"]::suspend()
[2019-10-26:10:53:36.577] qtmir.applications: Application["morph-browser"]::setInternalState(state=RunningInBackground)
[2019-10-26:10:53:36.578] qtmir.sessions: Wakelock released "3"
[2019-10-26:10:53:36.633] [PERFORMANCE]: Last frame took 49 ms to render.
[2019-10-26:10:53:45.415] [PERFORMANCE]: Last frame took 357 ms to render.
[2019-10-26:10:53:45.988] [PERFORMANCE]: Last frame took 306 ms to render.
[2019-10-26:10:55:52.826] qtmir.sensor: Screen::toggleSensors - enable= true
[2019-10-26:10:55:53.406] qtmir.surfaces: MirSurface[0x1afc220,"-"]::updateVisibility(true)
[2019-10-26:10:56:06.779] qtmir.applications: Application["morph-browser"]::setRequestedState(requestedState=running)
[2019-10-26:10:56:06.779] qtmir.applications: Application["morph-browser"]::resume()
[2019-10-26:10:56:06.780] qtmir.applications: Application["morph-browser"]::setInternalState(state=Running)
[2019-10-26:10:56:06.910] qtmir.sessions: Wakelock acquired "6"
[2019-10-26:10:56:06.911] qtmir.surfaces: MirSurface[0x1afc220,"-"]::updateVisibility(false)
[2019-10-26:10:56:06.958] [PERFORMANCE]: Last frame took 44 ms to render.
[2019-10-26:10:57:22.893] qtmir.sensor: Screen::onOrientationReadingChanged
[2019-10-26:10:57:22.899] qtmir.sensor: Screen::customEvent - new orientation Qt::ScreenOrientation(LandscapeOrientation) handled
[2019-10-26:10:59:06.461] qtmir.sensor: Screen::toggleSensors - enable= false
[2019-10-26:10:59:06.636] This backend doesn't support multiple users
[2019-10-26:10:59:06.672] qtmir.applications: Application["morph-browser"]::setRequestedState(requestedState=suspended)
[2019-10-26:10:59:06.673] qtmir.applications: Application["morph-browser"]::suspend()
[2019-10-26:10:59:06.674] qtmir.applications: Application["morph-browser"]::setInternalState(state=RunningInBackground)
[2019-10-26:10:59:06.677] qtmir.sessions: Wakelock released "6"
------------------------------>8
and in application-legacy-ubuntu-system-settings.log

8<------------------------------
...
library "/vendor/lib/egl/libGLESv2S3D_adreno.so" not found
xkbcommon: ERROR: Key "<LFSH>" added to modifier map for multile modifiers; Using Lock, ignoring shift
...
------------------------------>8

**Comment by Flaburgan** [comment]
I own a Fairphone 2 and, because of https://github.com/ubports/ubuntu-touch/issues/30, I have to restart it every morning. However, I never encountered that problem. Do you have more precise steps to reproduce?

**Comment by casimir7** [comment]
@Flaburgan: Unfortunately I cannot give more precise steps other than do a system restart (or shutdown and cold restart), wait until the Input field for unlocking the SIM begins to appear. It does not even finish the animation of sliding down to reveal the number block before it vanishes again and just the ubuntu touch splashscreen is displayed. After a few more seconds the locked desktop appears and the SIM cannot be unlocked again.
I just tried again, and again encountered the problem. Checked all the logs, no ERROR. I actually believe that the logs I gave above were created AFTER the upstart problem, so they might not actually have something to do with it. I do also have #30, by the way.
However, here's a screenshot of my network status after my last upstart. The button to unlock SIM2 (that's the slot I use) is greyed and disfunctional.
![screenshot20191028_160556200](https://user-images.githubusercontent.com/48692305/67690907-007f4300-f99e-11e9-8567-e6cca8a89bb0.png)

**Comment by Flaburgan** [comment]
As said, I never encountered that problem, but I use slot 1. Can you please try to reboot using slot 1, just to bisect the problem?

**Comment by casimir7** [comment]
Good idea! Could have thought of that myself.
If the SIM card is in slot 1, the problem did not occur during 4 restarts I tried. Although this is not a complete proof, it seems that the problem hence only appears if the (only) SIM card is in slot 2, which makes it far less a problem. Will edit my description of the bug above.

**Comment by Flaburgan** [comment]
Alright, so it would now be nice to know if this is specific to the fairphone or not. Can someone with another dual sim phone please try to reproduce?

**Comment by Danfro** [comment]
I missed this thread and created another issue. Dalton cleaned that up already. (Thanks)

So yes I can confirm this behaviour on E5 in edge and dev channel.
I do have two sim cards inserted.

Here for easier reference my findings:
It happens randomly, on average on every second reboot.
After some updates it does (almost) not happen at all.
On other days it needs up to 5 reboots until the sim pin dialog allows input

After installing the fix for #1277  it seems to be better. I does still skip the sim pin dialog, but I never needed more than a second attempt to be prompted for the sim pin dialog.

**Comment by buchnerh** [comment]
I think there a two discussions at ubports forum with the same bug:
https://forums.ubports.com/topic/3040/sometimes-sim1-is-blocked
https://forums.ubports.com/topic/3095/sim-locked-on-fairphone-2/5

**Comment by peat-psuwit** [comment]
The (merged) fix for #1277 fix seems to make reproduction harder. However, a reliable way to reproduce this problem is found: after getting the SIM lock screen, run `initctl restart unity8` from ADB shell.

**Comment by Flohack74** [comment]
Just to say had this case today one Oneplus also. It seems its not device-specific.

**Comment by luksus42** [comment]
I rebooted multiple times on FP2 with simcard in slot 2 on latest RC and could not reproduce this issue. Seems to be fixed.
